fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressBar.kt

```diff
@@ -0,0 +1,407 @@
+package com.microsoft.fluentui.tokenized.progress
+
+import androidx.compose.animation.core.*
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.compositionLocalOf
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.rotate
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.StrokeCap
+import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.unit.dp
+import com.microsoft.fluentui.theme.FluentTheme
+import com.microsoft.fluentui.theme.token.ControlTokens
+import com.microsoft.fluentui.theme.token.controlTokens.*
+import com.microsoft.fluentui.util.dpToPx
+import kotlin.math.absoluteValue
+
+val LocalProgressBarTokens = compositionLocalOf { ProgressBarTokens() }
+val LocalProgressBarInfo = compositionLocalOf { ProgressBarInfo() }
+
+@Composable
+fun getProgressBarTokens(): ProgressBarTokens {
+    return LocalProgressBarTokens.current
+}
+
+@Composable
+fun getProgressBarInfo(): ProgressBarInfo {
+    return LocalProgressBarInfo.current
+}
+
+/**
+ * Create a Determinate Linear Progressbar
+ *
+ * @param progress Progress of the progress indicator. 0.0 represents no progress and 1.0 represents full progress.
+ * @param progressbarHeight Optional width of the progress bar
+ * @param modifier Modifier for linear progress bar
+ * @param animationSpec Optional animation for the progress indicator. Look at [AnimationSpec]
+ * @param progressBarTokens Token values for linear progress bar
+ *
+ */
+@Composable
+fun LinearProgressBar(
```

Comment:
Check if with PM & designer, if "Indicator" as suffix could be used instead of "Bar"

Reply:
Break down the Linear, Circular & Shimmer in separate file.